File: testdata/bench/javascript/BenchmarkTest00082.js
// Safe: SSRF prevented via URL allowlist
const express = require('express');
const router = express.Router();
const ALLOWED_HOSTS = ['api.example.com', 'cdn.example.com'];
router.get('/test082', async (req, res) => {
    const url = req.query.url;
    const parsed = new URL(url);
    if (!ALLOWED_HOSTS.includes(parsed.hostname)) {
        return res.status(403).json({ error: 'Host not allowed' });
    }
    const response = await fetch(url);
    const data = await response.text();
    res.send(data);
});
module.exports = router;

File: testdata/bench/ruby/BenchmarkTest00082.rb
require 'net/http'
class WebhookController < ApplicationController
  ALLOWED_HOSTS = %w[api.example.com hooks.example.com].freeze
  def fetch
    url = params[:url]
    uri = URI.parse(url)
    return head(:forbidden) unless ALLOWED_HOSTS.include?(uri.host)
    response = Net::HTTP.get(uri)
    render plain: response
  end
end
